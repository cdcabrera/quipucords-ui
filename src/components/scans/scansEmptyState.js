import React from 'react';
import PropTypes from 'prop-types';
import {
  Button,
  EmptyState,
  EmptyStateBody,
  EmptyStateIcon,
  EmptyStatePrimary,
  EmptyStateVariant,
  Title
} from '@patternfly/react-core';
import { AddCircleOIcon } from '@patternfly/react-icons';
import { useNavigate } from '../router/routerContext';
import { useView } from '../view/viewContext';
import { useContextGetSources } from '../sources/sourcesContext';
import { helpers } from '../../common';
import { reduxTypes, storeHooks } from '../../redux';
import { translate } from '../i18n/i18n';

/**
 * Return a scans empty state.
 *
 * @param {object} props
 * @param {Function} props.t
 * @param {string} props.uiShortName
 * @param {Function} props.useContextGetSources
 * @param {Function} props.useDispatch
 * @param {Function} props.useNavigate
 * @param {Function} props.useView
 * @returns {React.ReactNode}
 */
const ScansEmptyState = ({
  t,
  uiShortName,
  useContextGetSources: useAliasContextGetSources,
  useDispatch: useAliasDispatch,
  useNavigate: useAliasNavigate,
  useView: useAliasView
}) => {
  const { viewId } = useAliasView();
  const dispatch = useAliasDispatch();
  const navigate = useAliasNavigate();
  const { totalResults, hasData } = useAliasContextGetSources();

  const onAddSource = () => {
    if (hasData) {
      navigate('/sources');
    } else {
      dispatch({
        type: reduxTypes.sources.CREATE_SOURCE_SHOW
      });
    }
  };

  return (
    <EmptyState className="quipucords-empty-state" variant={EmptyStateVariant.large}>
      <EmptyStateIcon icon={AddCircleOIcon} />
      <Title headingLevel="h1">
        {t('view.empty-state', { context: ['title', viewId], count: totalResults, name: uiShortName })}
      </Title>
      <EmptyStateBody>
        {t('view.empty-state', { context: ['description', viewId], count: totalResults })}
      </EmptyStateBody>
      <EmptyStatePrimary>
        <Button onClick={onAddSource}>
          {t('view.empty-state', { context: ['label', 'source-navigate'], count: totalResults })}
        </Button>
      </EmptyStatePrimary>
    </EmptyState>
  );
};

/**
 * Prop types
 *
 * @type {{uiShortName: string, useView: Function, t: Function, useContextGetSources: Function, useDispatch: Function,
 *     useNavigate: Function}}
 */
ScansEmptyState.propTypes = {
  t: PropTypes.func,
  uiShortName: PropTypes.string,
  useDispatch: PropTypes.func,
  useNavigate: PropTypes.func,
  useContextGetSources: PropTypes.func,
  useView: PropTypes.func
};

/**
 * Default props
 *
 * @type {{uiShortName: string, useView: Function, t: translate, useContextGetSources: Function, useDispatch: Function,
 *     useNavigate: Function}}
 */
ScansEmptyState.defaultProps = {
  t: translate,
  uiShortName: helpers.UI_SHORT_NAME,
  useDispatch: storeHooks.reactRedux.useDispatch,
  useNavigate,
  useContextGetSources,
  useView
};

export { ScansEmptyState as default, ScansEmptyState };
