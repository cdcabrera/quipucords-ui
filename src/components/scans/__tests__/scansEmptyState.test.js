import React from 'react';
import configureMockStore from 'redux-mock-store';
import { mount, shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import { ConnectedScansEmptyState, ScansEmptyState } from '../scansEmptyState';

describe('ScansEmptyState Component', () => {
  const generateEmptyStore = (obj = {}) => configureMockStore()(obj);

  it('should render a connected component', () => {
    const store = generateEmptyStore({ scansEmptyState: { sourcesExist: false } });
    const props = {};

    const component = shallow(
      <BrowserRouter>
        <ConnectedScansEmptyState {...props} />
      </BrowserRouter>,
      {
        context: { store }
      }
    );

    expect(component.find(ConnectedScansEmptyState)).toMatchSnapshot('connected');
  });

  it('should render a non-connected component', () => {
    const props = {
      sourcesExist: true
    };

    let component = mount(<ScansEmptyState {...props} />);
    expect(component.render()).toMatchSnapshot('non-connected exist');

    props.sourcesExist = false;
    component = mount(<ScansEmptyState {...props} />);
    expect(component.render()).toMatchSnapshot('non-connected do not exist');
  });
});
