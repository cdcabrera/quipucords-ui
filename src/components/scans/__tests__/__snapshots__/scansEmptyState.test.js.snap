// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScansEmptyState Component should render a basic component: basic 1`] = `
<EmptyState
  className="quipucords-empty-state"
  variant="lg"
>
  <EmptyStateHeader
    headingLevel="h1"
    icon={
      <EmptyStateIcon
        icon={[Function]}
      />
    }
    titleText={
      <React.Fragment>
        t(view.empty-state, {"context":"title","count":20,"name":"Quipucords"})
      </React.Fragment>
    }
  />
  <EmptyStateBody>
    t(view.empty-state, {"context":"description","count":20})
  </EmptyStateBody>
  <EmptyStateFooter>
    <EmptyStateActions>
      <Button
        onClick={[Function]}
      >
        t(view.empty-state_label, {"context":"source-navigate","count":20})
      </Button>
    </EmptyStateActions>
  </EmptyStateFooter>
</EmptyState>
`;

exports[`ScansEmptyState Component should render messaging if sources do not exist: do not exist 1`] = `
<EmptyState
  className="quipucords-empty-state"
  variant="lg"
>
  <EmptyStateHeader
    headingLevel="h1"
    icon={
      <EmptyStateIcon
        icon={[Function]}
      />
    }
    titleText={
      <React.Fragment>
        t(view.empty-state, {"context":"title","name":"Quipucords"})
      </React.Fragment>
    }
  />
  <EmptyStateBody>
    t(view.empty-state, {"context":"description"})
  </EmptyStateBody>
  <EmptyStateFooter>
    <EmptyStateActions>
      <Button
        onClick={[Function]}
      >
        t(view.empty-state_label, {"context":"source-navigate"})
      </Button>
    </EmptyStateActions>
  </EmptyStateFooter>
</EmptyState>
`;
